### PR TITLE
fuzzing :  fuzz testing support for oss-fuzz integration

### PR DESCRIPTION
**Pull request by 0x34d**
`Fuzz` testing support using [go-fuzz](https://go.dev/security/fuzz) .

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,3 +42,17 @@ test-coverage:
 
 clean:
 	rm -rf $(GIT_DIST_PATH)
+
+fuzz:
```
Comment (pjbgf) [inline Makefile:46]: Please add `make fuzz` to be executed on its own GitHub Action workflow on a [schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule).

File: Makefile [hunk 2]
```diff
@@ -42,3 +42,17 @@ test-coverage:
...
+	@go test -fuzz=FuzzDecode				$(PWD)/plumbing/object
+	@go test -fuzz=FuzzNewEndpoint			$(PWD)/plumbing/transport
+	@go test -fuzz=FuzzDecoder				$(PWD)/plumbing/protocol/packp
+	@go test -fuzz=FuzzDecoder				$(PWD)/plumbing/format/config
+	@go test -fuzz=FuzzPatchDelta			$(PWD)/plumbing/format/packfile
+	@go test -fuzz=FuzzDecodeFile			$(PWD)/utils/merkletrie/internal/fsnoder
+
+fuzzbug:
+	go test -fuzz=TestDecoder				$(PWD)/plumbing/protocol/packp
+	go test -fuzz=TestPatchDelta			$(PWD)/plumbing/format/packfile
```
Comment (pjbgf) [inline Makefile:58]: We can remove these and fix them as part of your follow-up PR.

**Review by pjbgf — CHANGES_REQUESTED** [review]
@0x34d thanks for working on this. :bow:

Please squash the commits and reword based on the [contributing guidelines](https://github.com/go-git/go-git/blob/master/CONTRIBUTING.md#format-of-the-commit-message).

File: Makefile (hunk 1 above)
Comment (0x34d) [reply Makefile:46]: > Please add make fuzz to be executed on its own GitHub Action workflow on a [schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule).


I have a better option. how about using `oss-fuzz`?

Comment (pjbgf) [reply Makefile:46]: That's a good shout. Please tag me on the PR to `oss-fuzz`.

**Review by pjbgf — APPROVED** [review]
@0x34d thanks for working on this. We can enable this for CI on a follow-up PR.

**Pull request merged**